### PR TITLE
Multi-line let body wrapping in expand-let (eigentrust pitfall #5)

### PR DESCRIPTION
**Pull request by kumavis**
Fixes pitfall **#5** from the 2026-04-23 eigentrust pitfalls memo.

## Summary

```prologos
(let [tnew := [eigentrust-step c p alpha t]]
  match [rat-lt [linf-norm [sub-vec tnew t]] eps]
    | true  -> tnew
    | false -> [eigentrust-iterate c p alpha eps [int- budget 1] tnew])
```

failed with `let: let with bracket bindings requires: (let [bindings…] body)`. The pitfall report attributed it to indent-handling. The actual root cause is different and instructive.

## Root cause

`let` lives inside `(...)` parens. parse-reader's `group-items` applies its "brackets win" rule and DROPS `indent-open` / `indent-close` markers around the body. So `(let [x := 1]\n  match x | _ -> x)` parses to `(let (x := 1) match x $pipe _ -> x)` — body tokens flat, length 7 instead of length 2 (bindings + body). The bracket-bindings branch in `expand-let` required exactly `(length rest) = 2` and rejected anything else.

## Fix

In `expand-let` branch 1 (bracket bindings), when `(length rest) > 2`, wrap the post-bindings tokens into a single body application. Localized, ~18 lines in `macros.rkt`. Avoids touching `parse-reader.rkt`'s "brackets suppress indent" rule (which is intentional for non-`let` forms — we don't want `(foo arg1\n  arg2)` to splice).

## Why the suggested fix would have been wrong

The pitfall's "suggested fix" (align WS `let` parser with the indent rule that `defn` / `match` honour) misdiagnoses. `defn` / `match` work because they're at top-level (no enclosing parens), where indent grouping IS active. Inside `(let ...)`, indent grouping is intentionally suppressed. The right fix is at the `let` macro, not at the parser.

## Test plan

- [x] New `tests/test-let-multiline-ws.rkt` — 11 cases: single-line baseline, multi-line with simple expression body, multi-line with nested `match` body (the eigentrust reproducer), edge cases
- [x] Regression: 243 tests across 8 files (let-multiline, let-arrow, eq-let-surface, process-ws-01, parse-reader, multi-body-defn, match-builtins, pattern-defn) all green in 37.8s
- [x] CI fix included (`benchmarks/micro/info.rkt`)

## Commits

1. `4eb61d1` — primary fix in `expand-let`
2. `c66a6d7` — CI fix (skip stale bench file)

https://claude.ai/code/session_01MbncYJnrvjzhbVWw4xGi5x

---
_Generated by [Claude Code](https://claude.ai/code/session_01MbncYJnrvjzhbVWw4xGi5x)_

### CODE REVIEW

File: racket/prologos/tests/test-let-multiline-ws.rkt [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#lang racket/base
+
+;;;
+;;; Regression tests for multi-line `let` forms in WS mode.
+;;;
+;;; Background — eigentrust pitfall #5 (2026-04-23) reported that
+;;;
+;;;     (let [tnew := [eigentrust-step c p alpha t]]
```
Comment (kumavis) [inline racket/prologos/tests/test-let-multiline-ws.rkt]: this is a little confusing as an example because it's using an unnecessary let block for multiple bindings with only a single binding. while this is allowed, it's not necessary. the block for bindings should be removed if it still produces the error.

**Review by kumavis — COMMENTED** [review]
hiero wants to review further

File: racket/prologos/data/cache/pnet/prologos/data/nat.pnet [hunk 2]
```diff
@@ -1 +1 @@
...dering () () () 2)) (lt-ord . #(struct:ctor-meta Ordering () () () 0)) (suc . #(struct:ctor-meta Nat () (Nat) (#t) 1)) (true . #(struct:ctor-meta Bool () () () 0)) (unit . #(struct:ctor-meta Unit () () () 0)) (zero . #(struct:ctor-meta Nat () () () 0))) #hasheq((Bool . (true false)) (Nat . (zero suc)) (Ordering . (lt-ord eq-ord gt-ord)) (Unit . (unit))) #hasheq() #hasheq(((Posit32 . Posit64) . #t) ((Nat . Rat) . #t) ((Posit8 . Posit16) . #t) ((Posit16 . Posit64) . #t) ((Int . Rat) . #t) ((Posit16 . Posit32) . #t) ((Nat . Int) . #t) ((Posit8 . Posit64) . #t) ((Posit8 . Posit32) . #t)) #hasheq() #hasheq() #hasheq() #hasheq() #hasheq() #hasheq() #hasheq() #hasheq() #hasheq((add . (x y)) (and . (a b)) (apply . (A B f x)) (bool-eq . (a b)) (bool-to-nat . (b)) (clamp . (low high)) (compose . (B C A g f x)) (const . (A B x _)) (double . (n)) (flip . (A B C f b a)) (ge? . (x y)) (gt? . (x y)) (id . (A x)) (implies . (a b)) (le? . (x y)) (lt? . (x y)) (max . (x y)) (min . (x y)) (mult . (x y)) (nand . (a b)) (nat-eq? . (x y)) (nor . (a b)) (not . (b)) (on . (B C A f g x y)) (or . (a b)) (pow . (base exp)) (pred . (n)) (sub . (x y)) (xor . (a b)) (zero? . (n))) #hasheq() #hasheq() #hasheq())
```
Comment (kumavis) [inline racket/prologos/data/cache/pnet/prologos/data/nat.pnet:1]: crazy diff detected. @hierophantos should "cache" be in `.gitignore` ?

**Review by hierophantos — APPROVED** [review]
LGTM. Surgical 18-line fix in `expand-let` Branch 1 (bracket-bindings format). Replaces the strict `(length rest) = 2` requirement with a cond that handles both the single-token body case (`(cadr rest)`) and the multi-token splice case (`(cdr rest)` wrapping post-bindings tokens as a body application).

Architectural discipline I appreciated: the PR explicitly debunks the "fix in parse-reader" hypothesis from the original pitfall report and explains why the parser-level fix would have been wrong — parse-reader's "brackets suppress indent" rule inside `(...)` parens is intentional for non-`let` forms (we don't want `(foo arg1\n  arg2)` to splice). The macro-level fix is the architecturally-correct location.

Test coverage strategic (11 cases): regressions for single-line / single-token-continuation / bracket-grouped bodies; the eigentrust reproducer (nested match body); multi-binding case; error-case regression (let with bindings but no body still raises).

Audit: no test asserts the old "let with bracket bindings requires" error; multi-line `(let [...] (let [...] ...))` patterns in `examples/2026-03-16-track4-acceptance.prologos:625-626` would benefit from this fix.

Mergeable as-is, no conflicts.

Approving.
